crates/topos/src/components/node/mod.rs

```diff
@@ -253,12 +270,26 @@ pub(crate) async fn handle_command(
                     shutdown(basic_controller, shutdown_trigger, shutdown_receiver).await;
                 }
                 Some(result) = processes.next() => {
-                    info!("Terminate: {result:?}");
-                    if let Err(e) = result {
-                        error!("Termination: {e}");
-                    }
                     shutdown(basic_controller, shutdown_trigger, shutdown_receiver).await;
                     processes.clear();
+                    match result {
+                        Ok(Ok(status)) => {
+                            if let Some(0) = status.code() {
+                                info!("Terminating with success error code");
+                            } else {
+                                info!("Terminating with error status: {:?}", status);
+                                std::process::exit(1);
+                            }
+                        }
+                        Ok(Err(e)) => {
```

Comment:
Same here for the shorter version

Reply:
Same compiler error as before - error must be of the same type to match such a pattern